<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="input_sensor_points_topic" default="/points_raw" description="Sensor points topic" />

  <arg name="param_file" default="$(find-pkg-share lidar_feature_library)/config/lidar_feature_library.param.yaml" />

  <arg name="output_colored_scan_topic" default="/colored_scan" description="Debug colored scan" />
  <arg name="output_curvature_scan_topic" default="/curvature_scan" description="Debug curvature scan" />
  <arg name="output_scan_edge_topic" default="/scan_edge" description="Scan edge features" />
  <arg name="output_scan_surface_topic" default="/scan_surface" description="Scan surface features" />

  <arg name="node_name" default="lidar_feature_extraction" />

  <node pkg="lidar_feature_extraction" exec="lidar_feature_extraction" name="$(var node_name)" output="log">
    <remap from="points_raw" to="$(var input_sensor_points_topic)" />

    <remap from="colored_scan" to="$(var output_colored_scan_topic)" />
    <remap from="curvature_scan" to="$(var output_curvature_scan_topic)" />
    <remap from="scan_edge" to="$(var output_scan_edge_topic)" />
    <remap from="scan_surface" to="$(var output_scan_surface_topic)" />

    <param from="$(var param_file)" />
  </node>

</launch>
